src/components/Asset/RelatedAssets/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { Asset } from '@oceanprotocol/lib'
+import { generateBaseQuery, queryMetadata } from '@utils/aquarius'
+import { useUserPreferences } from '@context/UserPreferences'
+import { SortTermOptions } from '../../../@types/aquarius/SearchQuery'
+import styles from './index.module.css'
+import { useCancelToken } from '@hooks/useCancelToken'
+import AssetList from '@shared/AssetList'
+
+export default function RelatedAssets({
+  tags,
+  nftAddress,
+  owner
+}: {
+  tags: string[]
+  nftAddress: string
+  owner: string
+}): ReactElement {
+  const { chainIds } = useUserPreferences()
+  const newCancelToken = useCancelToken()
+  const [relatedAssets, setRelatedAssets] = useState<Asset[]>()
+  const [isLoading, setIsLoading] = useState<boolean>()
+
+  function generateQuery(
+    size: number,
+    tagFilter: boolean,
+    ownerFilter: boolean
+  ) {
+    return {
+      chainIds,
+      esPaginationOptions: {
+        size
+      },
+      nestedQuery: {
+        must_not: {
+          term: { 'nftAddress.keyword': nftAddress }
+        }
+      },
+      filters: [
```

Comment:
why are you adding all these filters? you are duplicating them (index, purgatory, chainid)

Reply:
```
{
    "from": 0,
    "size": 3,
    "query": {
        "bool": {
            "must_not": {
                "term": {
                    "nftAddress.keyword": "0x0E122f7d453e983eF0233AF4E2AF4Cb053e5F075"
                }
            },
            "filter": [
                {
                    "terms": {
                        "chainId": [
                            1,
                            137,
                            56,
                            246,
                            1285
                        ]
                    }
                },
                {
                    "terms": {
                        "metadata.tags.keyword": [
                            "crypto",
                            "cryptocurrency",
                            "web3",
                            "data",
                            "sentiment",
                            "frequency-analysis"
                        ]
                    }
                },
                false,
                {
                    "term": {
                        "_index": "aquarius"
                    }
                },
                {
                    "term": {
                        "purgatory.state": false
                    }
                },
                {
                    "terms": {
                        "chainId": [
                            1,
                            137,
                            56,
                            246,
                            1285
                        ]
                    }
                },
                {
                    "term": {
                        "_index": "aquarius"
                    }
                },
                {
                    "term": {
                        "purgatory.state": false
                    }
                }
            ]
        }
    },
    "sort": {
        "stats.orders": "desc"
    }
}
```